fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+            stub_class_obj = stub_class(channel_object)
+            return stub_class_obj, channel_object
+        elif auth_type.lower().find('ssl') != -1 or auth_type.lower().find('tls') != -1:
+            log.info("Channel Authentication is secure : {}".format(auth_type))
+            assert certificate_path is not None, "Please specify the valid certificate path"
+            assert certificate_path.lower() != 'none', "Please specify the valid certificate path"
+            with open(certificate_path, 'rb') as certificate:
+                channel_cred = grpc.ssl_channel_credentials(root_certificates=certificate.read())
+            channel_object = grpc.secure_channel(channel, channel_cred)
+            stub_class_obj = stub_class(channel_object)
+            return stub_class_obj, channel_object
+        else:
+            raise NotImplementedError("Not Implemented these {} Authentication types ".format(auth_type))
+
+    @function("function_grpc")
+    def _function_grpc_function(self, event, *args, **kwargs):
+        """Function: A function to communicate with GRPC Servers based on the Communication Methods Specified."""
+        try:
+            # Get the function parameters:
+            grpc_channel = kwargs.get("grpc_channel")  # text
+            grpc_function = kwargs.get("grpc_function")  # text
+            grpc_function_data = kwargs.get("grpc_function_data")  # text
+
+            log = logging.getLogger(__name__)
+            log.info("grpc_channel: %s", grpc_channel)
+            log.info("grpc_function: %s", grpc_function)
+            log.info("grpc_function_data: %s", grpc_function_data)
+
+            yield StatusMessage("starting...")
+            response_received = None
+            grpc_function_json_data = None
+
+            # Parsing the variable data from config file and resilient inputs
+            try:
+                _grpc_package_name = grpc_function.split(':')[0].strip()
+                _grpc_rpc_stub_method_name = grpc_function.split(':')[1].split("(")[0].strip()
+                _grpc_request_method_name = re.sub("\)", '', grpc_function.split(':')[1].split("(")[1])
+                _grpc_request_method_name = _grpc_request_method_name.strip()
+                config_file_data = self.options.get(_grpc_package_name).split(',')
+                _grpc_secure_connection = config_file_data[1]
+                _grpc_communication_type = config_file_data[0]
+                _grpc_google_tocken = config_file_data[2]
+                _grpc_certificate_path = config_file_data[2]
+                _grpc_interface_file_dir = self.options.get('interface_dir')
+
+                log.info("gRPC Package Name : {} gRPC Stub Method Name : {} gRPC Request Class Name : {}" \
+                         .format(_grpc_package_name, _grpc_rpc_stub_method_name, _grpc_request_method_name))
+            except Exception as e:
+                yield StatusMessage("failed to parse the function parameters.{}".format(e))
```

Comment:
Rather than `yield StatusMessage("...")` and `raise FunctionError(e)`, would could just:
```
raise ValueError("msg...")
```

Reply:
ok